doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,498 @@
+         :compact:
+
+   .. group-tab:: ``cmake``
+
+      .. zephyr-app-commands::
+         :tool: cmake
+         :app: share/sysbuild
+         :board: reel_board
+         :goals: build
+         :gen-args: -DAPP_DIR=samples/hello_world -DSB_CONF_FILE=sysbuild-mcuboot.conf
+         :compact:
+
+.. _sysbuild_zephyr_application:
+
+Adding Zephyr applications to sysbuild
+**************************************
+
+You can use the ``ExternalZephyrProject_Add()`` function to add Zephyr
+applications as sysbuild domains. Call this CMake function from your main
+:file:`CMakeLists.txt` file, or any other CMake file you know will run.
+
+Targeting the same board
+========================
+
+To include ``my_sample`` as another sysbuild domain, targeting the same board
+as the main image, use this example:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     BUILD_ALWAYS True
+   )
+
+This could be useful, for example, if your board requires you to build and flash an
+SoC-specific bootloader along with your main application.
+
+Targeting a different board
+===========================
+
+For example, if your main application is built for ``mps2_an521``, and your
+helper application must target the ``mps2_an521_remote`` board (cpu1), add
+a CMake function call that looks like this:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     BOARD mps2_an521_remote
+     BUILD_ALWAYS True
+   )
+
+This could be useful, for example, if your main application requires another
+helper Zephyr application to be built and flashed alongside it, but the helper
+runs on another core in your SoC.
+
+Conditionally, with Kconfig
+===========================
+
+You can control whether extra applications are included as sysbuild domains
+using Kconfig.
+
+If the extra application image is special for the board or an application, then
+it is possible to create to additional files, a :file:`sysbuild.cmake` and a
+:file:`Kconfig.sysbuild` file.
+
+For an application, this would look like this:
+
+.. code-block:: none
+
+   <home>/application
+   ├── CMakeLists.txt
+   ├── prj.conf
+   ├── Kconfig.sysbuild
+   └── sysbuild.cmake
+
+Here, :file:`sysbuild.cmake` contains:
+
+.. code-block:: cmake
+
+   if(SB_CONFIG_SECOND_SAMPLE)
+     ExternalZephyrProject_Add(
+       APPLICATION second_sample
+       SOURCE_DIR <path-to>/second_sample
+       BUILD_ALWAYS True
+     )
+   endif()
+
+And :file:`Kconfig.sysbuild` contains:
+
+.. code-block:: kconfig
+
+   source "sysbuild/Kconfig"
+
+   config SECOND_SAMPLE
+           bool "Second sample"
+           default y
+
+This will include ``second_sample`` by default, while still allowing you to
+disable it using the Kconfig option ``SECOND_SAMPLE``. For more information on
+setting sysbuild Kconfig options, see :ref:`sysbuild_kconfig_namespacing`.
```

Comment:
```suggestion
This will include ``second_sample`` by default, while still allowing you to
disable it using the Kconfig option ``SECOND_SAMPLE``. 

For more information on setting sysbuild Kconfig options,
see :ref:`sysbuild_kconfig_namespacing`.
```